providers/ldap/ldap_client.go

```diff
@@ -110,7 +110,8 @@ func (l *LClient) newConn() (*ldap.Conn, error) {
 			certFound = true
 		}
 		if certFound {
-			tlsConfig = &tls.Config{RootCAs: pool}
+			tlsConfig = &tls.Config{RootCAs: pool, InsecureSkipVerify: false, ServerName: l.Config.Server}
+			tlsConfig.BuildNameToCertificate()
```

Comment:
@cjellick This isn't necessary, as per the definition here https://golang.org/pkg/crypto/tls/#Config.BuildNameToCertificate
It's only used if we set the Certificates field of tls config which we don't